sdk/formrecognizer/Azure.AI.FormRecognizer/src/ClientCommon.cs

```diff
@@ -35,6 +36,18 @@ public static Guid ValidateModelId(string modelId, string paramName)
             return guid;
         }
 
+        public static string GetResponseHeader(ResponseHeaders responseHeaders, string headerName)
+        {
+            if (responseHeaders.TryGetValue(headerName, out var headerValue))
+            {
+                return headerValue;
+            }
+            else
+            {
+                throw new KeyNotFoundException($"Header '{headerName}' was not present in the response sent by the server.");
```

Comment:
Unlikely to happen, but I added this check so we don't end up with a meaningless `NullReferenceException` in case of a service-side issue.